File: backend/masz/Controllers/api/v1/Enums/AutoModerationActionEnumController.ts
using System;
using System.Collections.Generic;
using masz.Dtos.Enum;
using masz.Models;
using masz.Translations;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;

namespace masz.Controllers
{
    [ApiController]
    [Route("api/v1/enums/")]
    public class AutoModerationActionEnumController : SimpleController
    {
        private readonly ILogger<AutoModerationActionEnumController> logger;

        public AutoModerationActionEnumController(ILogger<AutoModerationActionEnumController> logger, IServiceProvider serviceProvider) : base(serviceProvider)
        {
            this.logger = logger;
        }

        [HttpGet("automodaction")]
        public IActionResult AutoModActions([FromQuery] Language? language = null) {
            logger.LogInformation($"{HttpContext.Request.Method} {HttpContext.Request.Path} | Incoming request.");

            List<EnumDto> enums = new List<EnumDto>();
            enums.Add(new EnumDto((int) AutoModerationAction.None, translator.T(language).EnumsAutoModActionsNone()));
            enums.Add(new EnumDto((int) AutoModerationAction.ContentDeleted, translator.T(language).EnumsAutoModActionsContentDeleted()));
            enums.Add(new EnumDto((int) AutoModerationAction.CaseCreated, translator.T(language).EnumsAutoModActionsCaseCreated()));
            enums.Add(new EnumDto((int) AutoModerationAction.ContentDeletedAndCaseCreated, translator.T(language).EnumsAutoModActionsContentDeletedAndCaseCreated()));

            return Ok(enums);
        }
    }
}

File: backend/MASZ/Controllers/api/v1/Enums/AutoModerationActionEnumController.cs
using MASZ.Dtos.Enum;
using MASZ.Enums;
using Microsoft.AspNetCore.Mvc;

namespace MASZ.Controllers
{
    [ApiController]
    [Route("api/v1/enums/")]
    public class AutoModerationActionEnumController : SimpleController
    {
        public AutoModerationActionEnumController(IServiceProvider serviceProvider) : base(serviceProvider)
        {
        }

        [HttpGet("automodaction")]
        public IActionResult AutoModActions([FromQuery] Language? language = null)
        {
            _translator.SetContext(language);
            return Ok(new List<EnumDto>()
            {
                EnumDto.Create((int) AutoModerationAction.None, _translator.T().Enum(AutoModerationAction.None)),
                EnumDto.Create((int) AutoModerationAction.ContentDeleted, _translator.T().Enum(AutoModerationAction.ContentDeleted)),
                EnumDto.Create((int) AutoModerationAction.CaseCreated, _translator.T().Enum(AutoModerationAction.CaseCreated)),
                EnumDto.Create((int) AutoModerationAction.ContentDeletedAndCaseCreated, _translator.T().Enum(AutoModerationAction.ContentDeletedAndCaseCreated)),
                EnumDto.Create((int) AutoModerationAction.Timeout, _translator.T().Enum(AutoModerationAction.Timeout))
            });
        }
    }
}